combat-trainer.lic

```diff
@@ -926,13 +950,22 @@ class SpellProcess
     @buff_force_cambrinth = settings.combat_trainer_buffs_force_cambrinth
     echo("  @buff_force_cambrinth: #{@buff_force_cambrinth}") if $debug_mode_ct
 
+    @regalia_array = settings.cycle_armors_regalia
+    echo("  @regalia_array: #{@regalia_array}") if $debug_mode_ct
+
+    @regalia_spell = settings.waggle_sets['regalia']['Regalia']
+    echo("  @regalia_spell: #{@regalia_spell}") if $debug_mode_ct
+    
     @perc_health_timer = Time.now
 
     @symbiosis_learning_threshold = settings.symbiosis_learning_threshold
     echo(" @symbiosis_learning_threshold: #{@symbiosis_learning_threshold}") if $debug_mode_ct
 
-    Flags.add('ct-spelllost', 'Your pattern dissipates with the loss of your target')
+    Flags.add('ct-spelllost', 'Your pattern dissipates with the loss of your target', 'Your concentration lapses')
     Flags.add('ct-need-bless', ' passes through the .* with no effect')
+    Flags.add('ct-starlight-depleted', 'enough starlight') if DRStats.trader?
+    Flags.add('ct-regalia-expired', 'Your .* glimmers weakly') if DRStats.trader?
+    Flags.add('ct-cancel-regalia')
```

Comment:
❓ What is the matching text for this Flag?

Reply:
I struggled with this one, and now I see a better way to do it.  The existing logic for armor-swapping (and thus also regalia) is up in SetupProcess.  So when SpellProcess detects that you've run out of starlight and can't cast regalia anymore, it needs to signal SetupProcess to stop swapping around and switch to your standard equipment set.  To do that I manually toggled a flag, which seems stupid in retrospect.

Probably better to add another attribute to gamestate and have the armor logic check for that.  I'll make a change, remove the flag and its deletion.

Reply:
👍

Reply:
Done and done.  Now uses game_state.regalia_cancel to trigger regalia_shutdown, which will remove everything, swap back to normal gear, and remove regalia from training.